java/src/main/java/system/gameplay/roleplay/DiceRoller.java

```diff
@@ -5,26 +5,53 @@
 import java.util.List;
 import java.util.Random;
 
-public class DiceRoller {
+public final class DiceRoller {
 
-    private final Random random;
+    private static final Random random = new Random();
 
-    public DiceRoller() {
-        random = new Random();
+    public static List<Integer> rollAbilityScores() {
+        //TODO: (A) This line is here so the compiler won't complain, remove it when you add the functionality for this method
+        ArrayList<Integer> rollAbilityScoresList = new ArrayList<>();
+        int topThreeSixSidedDie = topThreeSixSidedDie();
+        for (int i = 1; i <= 6; i++) {
+                rollAbilityScoresList.add(topThreeSixSidedDie);
+                topThreeSixSidedDie = topThreeSixSidedDie();
+        }
+
+        // Sort scores to make it easier for player to keep track of how they're prioritizing their abilities
+        sortIntsHighToLow(rollAbilityScoresList);
+        return rollAbilityScoresList;
     }
 
-    public List<Integer> rollAbilityScores() {
-        //TODO: (A) This line is here so the compiler won't complain, remove it when you add the functionality for this method
-        return new ArrayList<>();
+    public static int roll6() {
+        return random.nextInt(6) + 1;
     }
 
-    public int roll20() {
-        // Returns a random int between 1 and 20
+    public static int roll20() {
         return random.nextInt(20) + 1;
     }
 
-    private List<Integer> sortIntsHighToLow(List<Integer> list) {
-        Collections.reverse(list);
-        return list;
+    public static int topThreeSixSidedDie() {
+        ArrayList<Integer> diceRollValues = new ArrayList<>();
+        for (int i = 0; i < 4; i++) {
+            diceRollValues.add(roll6());
+        }
+
+        sortIntsHighToLow(diceRollValues);
+
+        diceRollValues.remove(3);
+        int diceRollValuesSum = 0;
+
+        for (int i: diceRollValues) {
+            diceRollValuesSum += i;
+        }
+
+        return diceRollValuesSum;
+    }
+
```

Comment:
Copy paste this TODO for me:

```java
// TODO: Maybe move this method out into a seperate utility class that will deal with common list functionality
```

This is probably something that we'll use in other places, if we do then we should keep it in a class that deals with Lists, rather than the `DiceRoller`. It would be bad design to go to the `DiceRoller` in order to sort a list, or do anything that isn't directly related to rolling dice for that matter. For now, since this method is private, it's totally fine that it's in here though. The main reason I made it a private method here rather than putting it into a util class to begin with was because I wanted you to see it so that you could use it to complete the issue. Tbh, this probably doesn't even need to be a method now that you've seen it and know how to sort a list in the desired fashion, but it does make the code more readable so we might as well keep it.